### DISCUSSION TITLE
Ruby version 3?

### DISCUSSION THREAD

**Discussion by bcollierjones** [sqids/sqids-ruby#2] [Ideas]
Any chance of support for Ruby 3.0? Unfortunately, we can't use ruby 3.1 just yet.

**Comment by 4kimov** [comment]
Try v0.1.1?

**Reply by bcollierjones** [reply]
Initial tests seem to be working great. Thank you so much! We've been using hashids on a project in development and definitely want to be using the latest and greatest before we go into production.

**Reply by 4kimov** [reply]
Good, happy to hear! Good luck with production and please re-open if you see any issues.

**Comment by bcollierjones** [comment]
This is probably unrelated to Ruby, and might be purely cosmetic, so I can put this in a new discussion if necessary or just ignore it because it is working as intended.

I wrote a small script to see how the length of sqids change compared to hash ids. Ideally we'd like to pick a minimum length that will stay relatively consistent and only grow longer late into the product lifecycle. However, while sqids definitely produce more random batches of characters, the length of the sqids swings back and forth.  Here is sample code and output to demonstrate:

### Sample Output
```
Number : sqid / hashid
       0 :           bV /           gY
      44 :           NL /          gJY
      59 :          zDP /          ADO
    1936 :          Ho1 /         31RR
    3481 :         nbZZ /         4xM6
   14777 :       jhBJPv /         Z0x2
   14778 :         YPC3 /         14O0
   17420 :       A5QBCu /         JPGJ
   17421 :         UBFi /         K6KM
   26085 :       Cyrgqg /         8Wxj
   26086 :         kjyB /         9GyJ
   33404 :       wBjuT9 /         mkME
   33405 :         LY1u /         nlNY
   50058 :       ZXbVGW /         zgWO
   50059 :         kcxC /         AEvp
   63620 :       3nq8z8 /         Jo0J
   63621 :         nUpG /         KpJM
   73642 :       wBjku9 /         9JrZ
   73643 :         LmYu /         0wRK
   73654 :       A5QkfF /         vn2X
   73655 :         Uv2S /         wopM
   85184 :         ys5N /        73888
  126755 :       kLpWV2 /        wplAJ
  126756 :         mpHH /        xkmBJ
```
### Code
```ruby

def test
  sqid_encoder = Sqids.new
  hashid_encoder = Hashids.new
  sqid_size = 0
  hashid_size = 0
  i = 0

  1_000_0000.times do |i|
    new_sqid = sqid_encoder.encode([i])
    new_hashid = hashid_encoder.encode([i])

    if new_sqid.size != sqid_size || new_hashid.size != hashid_size
      puts "%8d : %12s / %12s" % [i, new_sqid, new_hashid]
      sqid_size = new_sqid.size if new_sqid.size != sqid_size
      hashid_size = new_hashid.size if new_hashid.size != hashid_size
    end
  end
end
```

**Reply by 4kimov** [reply]
Thanks for the sample code. Yes, it does go back and forth. My guess is the outliers are hits on the blocklist - meaning the generated ID matches either fully or partly on the [default blocklist](https://github.com/sqids/sqids-spec/blob/main/src/blocklist.json), and an attempt to re-generate the ID results in a longer one (you can test that by providing a blank blocklist which would disable that feature). Not really an easy fix, so my only suggestion for now would probably be to pick a longer `minLength` (eg: anything between 6 and 12 depending on your requirements), or alternatively disable blocklist.
